crates/apub/src/protocol/objects/page.rs

```diff
@@ -167,11 +167,21 @@ impl Page {
 }
 
 impl Attachment {
-  pub(crate) fn new(url: DbUrl) -> Attachment {
-    Attachment::Link(Link {
-      href: url.into(),
-      r#type: Default::default(),
-    })
+  /// Creates new attachment for a given link and mime type.
+  pub(crate) fn new(url: Url, media_type: Option<String>) -> Attachment {
+    let is_image = media_type.clone().unwrap_or_default().starts_with("image");
```

Comment:
Can you use the mime type library you added for this now?

Reply:
Its possible, but because its an option its a bit complicated to do the parsing. The string comparison works just fine so I dont see any reason to make it more complicated.